File: 007 Reverse Integer/ReverseInteger.java
class Solution {
    public int reverse(int x) {
        int result = 0;        
        int remainder;
        int quotient = x;
        while (quotient != 0) {
            remainder = quotient % 10;
            quotient = quotient / 10;       
            if (result > Integer.MAX_VALUE / 10 || result == Integer.MAX_VALUE / 10 && remainder > 7) {
                return 0;
            }
            if (result < Integer.MIN_VALUE / 10 || result == Integer.MIN_VALUE / 10 && remainder < -8) {
                return 0;
            }
            result = result * 10 + remainder;
        }        
        
        return result;
    }
}

File: 007 Reverse Integer/reverse_integer.py
class Solution:
    def reverse(self, x: int) -> int:
        negative = False
        if (x < 0):
            x = -x
            negative = True
        res = 0
        while x != 0 :
            remainder = x % 10
            x = x // 10
            res = res * 10 + remainder
            
        if negative:
            res = -res
            
        if -2147483648 <= res <= 2147483647:
            return res
        else:
            return 0